package.json

```diff
@@ -0,0 +1,22 @@
+{
+  "name": "eclipse-spectrum-theme-dev",
+  "version": "1.0.0",
+  "description": "Development utilities for Eclipse Spectrum Theme",
+  "main": "index.js",
+  "scripts": {
+    "format": "./node_modules/prettier/bin-prettier.js --write ."
```

Comment:
@ingomohr in theory this could run with a github action which would amend the previous commit when merging to master. The downside is that the formatting changes wouldn’t be visible to the reviewer until the PR is merged.

Reply:
> @ingomohr in theory this could run with a github action which would amend the previous commit when merging to master. The downside is that the formatting changes wouldn’t be visible to the reviewer until the PR is merged.

@AObuchow Hm, yes. I see the upside on this one - and also the same downside. :) - I image some case in which the committer added some extra empty line and some comment to a certain line to point out some thing, and then the commit is just being replaced by another commit that automatically changed format and - from the committer's perspective - "scrambled" his comments. And they're attempting to add to the PR "I've added a comment to line 99 - er... line 102... and - wait, what is this? This isn't what I actually committed..." ... something in that direction. ;)
Also, an amend w/ what user? Wouldn't this replace the author information of the commit?

Still, I agree that extra commits would also not look that nice in the commit history. But (another updside to them) they would increase readability of "who changed what with which commit?".

Just thinkin'... :)

Reply:
Thanks @ingomohr these are very good points.
I decided for now to drop the idea of autoformatting on every commit.

Instead, the contributor has a choice to either format via the CLI by running `npm install` (to download prettier) and then `npm run format`, or by an Eclipse build action which I've included in this PR.

In the future, auto-formatting could be suitable for releases perhaps, as a kind of "checkpoint". This would create a new commit so that the formatting changes are easily observable. However, if contributors frequently format their PR's with prettier, this shouldn't really be necessary.